catalog/rocks/staging.go

```diff
@@ -0,0 +1,78 @@
+package rocks
+
+import (
+	"context"
+
+	"github.com/treeverse/lakefs/db"
+	"github.com/treeverse/lakefs/logging"
+)
+
+type PGStagingManager struct {
+	db  db.Database
+	log logging.Logger
+}
+
+func NewStageManager(db db.Database) *PGStagingManager {
+	return &PGStagingManager{
+		db:  db,
+		log: logging.Default().WithField("service_name", "postgres_staging_manager"),
+	}
+}
+
+func (p *PGStagingManager) GetEntry(ctx context.Context, st StagingToken, path Path) (*Entry, error) {
+	res, err := p.db.Transact(func(tx db.Tx) (interface{}, error) {
+		entry := &Entry{}
+		err := tx.Get(entry, "SELECT address, last_modified_date, size, e_tag, metadata FROM staging_entries WHERE staging_token=$1 AND path=$2", st, path)
+		return entry, err
+	}, p.txOpts(ctx, db.ReadOnly())...)
+	if err != nil {
+		return nil, err
+	}
+	entry := res.(*Entry)
+	if entry.Address == "" {
+		return nil, nil
+	}
```

Comment:
so the part I was worried about here is the hidden assumption that I'm not using empty Address for non-tombstone